### ISSUE TITLE
PWC is missing country

### ISSUE THREAD

**Issue by rggelles** [georgetown-cset/parat#239]
It looks like this is an issue in country aggregation; booz allen is getting aggregated to pwc even though pwc isn't in the organizations table? Not sure if it was there and was deleted or if something is wrong with our aggregation but we should check on this.

**Comment by rggelles** [comment]
Okay, checking this out in our data, this is because Booz Allen is listed as a child of PWC in our data and PWC is listed as having a CSET_id (probably its historical murdic_id) but PWC isn't in our dataset.

I checked on Booz Allen and it looks like only one arm of it was actually acquired by PWC, which is probably why it's in our dataset and PWC isn't. My guess is the solution here is to remove PWC as a parent and then re-ingest our data (although I'm confirming this with Ngor).

However, there are a few downstream implications here, so I want to check in before doing this:

1. @za158 did you end up consolidating the PARAT data in airtable? And if so, have you made modifications to it? And have we modified our Airtable-ingest pipelines to accomodate these consolidations? If I'm going to re-ingest, I'll either need to use the old data or use your new data; ideally if you've made updates I'd make any fixes in the new data but if I need to modify the Airtable-ingest pipeline first that's another step here.
2. @jmelot If we need to update this data we'll also need to rerun all the paper stats data eventually. Before we do that we need a solution to the cost problem in the all-papers part of that; I'm not sure if you've had any other thoughts here. I know you had some really cool ideas but I recall your most recent one ended up timing out/complaining that we'd tricked BigQuery to save money? Any thoughts on dealing with that? The alternate solution is we get all the fixes in and run it once with the year limiters in, which would be expensive but not prohibitive.

**Comment by rggelles** [comment]
Got confirmation with Ngor that the step forward here is to remove PWC as a parent from Booz Allen (Booz Allen split and only half of it was acquired, so we still want the non-acquired half in our data).

So I just need answers from you guys on steps forward for actually making this change, and this can be fixed!

**Comment by za158** [comment]
> @za158 did you end up consolidating the PARAT data in airtable? And if so, have you made modifications to it?

Yes - we have consolidated PARAT ER data in airtable (https://airtable.com/appqtfVHf5tIRsfkI/tbl12gdOy2YSf0whH/viwhrBi4UHZtwbHv9?blocks=hide) and are making any further edits there (this is also what we're using to do the ongoing regex audit). I will get rid of the parent-child relationship for PWC in the consolidated data.

> And have we modified our Airtable-ingest pipelines to accomodate these consolidations?

My understanding is that the pipelines will be modified to read from the consolidated table, and this may have already happened, but I will need to defer to @jmelot there

**Comment by rggelles** [comment]
Sounds good! Thanks for making that fix. Will await update from @jmelot on integration via pipeline.

**Comment by jmelot** [comment]
I haven't updated the airtable pipeline to reflect the new base. I had assumed that was on my to-do list but if you want to do it, go for it!

I also haven't worked on the performance issues recently. I still believe that they're tractable because:

1. we should only need to run each regex on each raw org name once to determine whether a company-raw org name match exists
2. if it's expensive to do ^ in time or money, we can cache the results and only run the regex on unseen raw org names

Because BQ complained about resource use when I tried to do (1) my plan was to write a beam script to do it and get a cost estimate. I haven't done that yet, but it should be straightforward.

**Comment by rggelles** [comment]
If you have time to do the pipeline updates, I am happy to leave that on your plate! If you need me to take it on, let me know. I just wanted to make sure everything needed to finish the PWC-country issue was done.

Beam script sounds like a good solution. In that case, I'll leave that for you as well, since it sounds like you have a plan.

**Comment by rggelles** [comment]
This is fixed! PWC is no more and Booz Allen has a country now.
